great_expectations/datasource/datasource_dict.py

```diff
@@ -129,13 +129,7 @@ def _init_fluent_datasource(self, name: str, ds: FluentDatasource) -> FluentData
                         datasource_name=name,
                         data_asset_name=asset.name,
                     )
-                    cached_data_asset = self._in_memory_data_assets.get(in_memory_asset_name)
```

Comment:
This code block is removed because we no longer store the dataframe on the asset.